File: 0073-set-matrix-zeroes/solution.c
void setZeroes(int** matrix, int matrixSize, int* matrixColSize) 
{
    int *row=(int *)malloc(matrixSize* sizeof(int));
    int *col=(int *)malloc(matrixColSize[0] * sizeof(int));
    for(int i=0;i<matrixSize;i++)
    {
        row[i]=0;
    }

    for(int i=0;i<matrixColSize[0];i++)
    {
        col[i]=0;
    }

    for(int i=0;i<matrixSize;i++)
    {
        for(int j=0;j<matrixColSize[0];j++)
        {
            if(matrix[i][j]==0)
            {
                row[i]=1;
                col[j]=1;
            }
        }
    }    
    for(int i=0;i<matrixSize;i++)
    {
        for(int j=0;j<matrixColSize[0];j++)
        {
            if(row[i]==1 || col[j]==1)
            {
                matrix[i][j]=0;
            }
        }
    }
}

File: 0073-set-matrix-zeroes/solution.java
class Solution {
    public void recur1(int row,int col,int[][] mat,int c){
        if(col>=c){
            return ;
        }
        mat[row][col]=0;
        recur1(row,col+1,mat,c);
    }
    public void recur2(int row,int col,int[][] mat,int r){
        if(row>=r){
            return ;
        }
        mat[row][col]=0;
        recur2(row+1,col,mat,r);
    }
    public void setZeroes(int[][] mat) {
        List<int[]> list = new ArrayList<>();
        int r=mat.length;
        int c=mat[0].length;
        for(int i=0;i<r;i++){
            for(int j=0;j<c;j++){
                if(mat[i][j]==0)
                    list.add(new int[] {i,j});
            }
        }   
        for(int[] cur:list){
            int i=cur[0];
            int j=cur[1];
            recur1(i,0,mat,c);
            recur2(0,j,mat,r);
        }
    }
}
